packages/react-dom/src/server/ReactPartialRenderer.js

```diff
@@ -1206,21 +1216,21 @@ class ReactDOMServerRenderer {
       if (selectValue != null) {
         let value;
         if (props.value != null) {
-          value = props.value + '';
+          value = props.value;
```

Comment:
Why is this unnecessary now?

Reply:
I'll double check. I removed a lot of string conversion to avoid stringifying symbols.

Reply:
(really, I replaced it with `ToStringValue` method, but then stripped them away to see if they were actually necessary)

Reply:
We should make sure this doesn't break number/string comparisons.

Reply:
Good catch. It did. I've added test coverage and fixed this by adding a safe string comparison method. I like this approach over eagerly coercing to strings because it makes it clearer when it's necessary.

Reply:
(this was in 7619d04c516cb40eab31362164c320f7f057dd89)